// gatsby-source-datocms/hooks/sourceNodes/createTypes/item/fields/link.js
"use strict";

var _require = require('humps'),
    camelize = _require.camelize,
    pascalize = _require.pascalize;

module.exports = function (_ref) {
  var parentItemType = _ref.parentItemType,
      field = _ref.field,
      schema = _ref.schema,
      gqlItemTypeName = _ref.gqlItemTypeName,
      entitiesRepo = _ref.entitiesRepo;
  var fieldKey = camelize(field.apiKey);
  var parentItemTypeName = gqlItemTypeName(parentItemType);
  var itemTypeIds = field.validators.itemItemType.itemTypes;

  if (itemTypeIds.length === 0) {
    return {
      fieldType: 'String'
    };
  }

  if (itemTypeIds.length === 1) {
    var linkedItemType = entitiesRepo.findEntity('item_type', itemTypeIds[0]);
    return {
      fieldType: {
        type: gqlItemTypeName(linkedItemType),
        resolve: function resolve(parent, args, context) {
          var id = 'locale' in parent && 'value___NODE' in parent ? parent.value___NODE : parent["".concat(fieldKey, "___NODE")];

          if (id) {
            return context.nodeModel.getNodeById({
              id: id
            });
          }
        }
      }
    };
  }

  var unionType = "DatoCmsUnionFor".concat(parentItemTypeName).concat(pascalize(field.apiKey));
  var unionTypes = itemTypeIds.map(function (id) {
    return gqlItemTypeName(entitiesRepo.findEntity('item_type', id));
  });
  return {
    types: [schema.buildUnionType({
      name: unionType,
      types: unionTypes
    })],
    fieldType: {
      type: unionType,
      resolve: function resolve(parent, args, context) {
        var id = 'locale' in parent && 'value___NODE' in parent ? parent.value___NODE : parent["".concat(fieldKey, "___NODE")];

        if (id) {
          return context.nodeModel.getNodeById({
            id: id
          });
        }
      }
    }
  };
};
